target_bigquery.py

```diff
@@ -355,7 +352,23 @@ def write_rows_to_bigquery(streams, emit_state_after_write=False):
                                 fixed_rows[number_of_rows // 2 :],
                                 row_ids=ids[number_of_rows // 2 :],
                             )
-                        elif getattr(e, "errors", [{}])[0].get("reason") in RETRYABLE_ERROR_CODES:
+
+                        if (
+                            "payload size exceeds the limit" in error_string
+                            or "too many rows present" in error_string
+                        ):
+                            insert_in_halves()
+                        # While this is similar to the above, we want to avoid doing the json.dumps
+                        # if possible, so will only do it if we didn't get the expected errors
+                        elif len(json.dumps(fixed_rows)) > 10000000:
+                            insert_in_halves()
+                        elif (
+                            # Normally there's an `errors` attribute with a list of errors on the
+                            # exception object if it comes from Google, but sometimes it's empty or
+                            # we get a native exception which doesn't have the `errors` attribute
+                            next(iter(getattr(e, "errors", [{}])), {}).get("reason")
+                            in RETRYABLE_ERROR_CODES
```

Comment:
this isn't very readable, surely you can do something like
```
errors = getattr(e, "errors", [])
.
.
.
elif errors and errors[0].get(reason):
```
if errors is an empty list it evaluates to `False` and the execution stops, makes it a lot more readable

Reply:
Right, empty arrays are truthy in Javascript so I missed that trick!